src/PasswordPolicy.php

```diff
@@ -0,0 +1,289 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getRole(): RoleInterface {
+    return $this->role;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpireSeconds(int $seconds): PasswordPolicyInterface {
+    if ($seconds < 0) {
+      throw new PolicyInvalidArgumentException('The expiry given in seconds must be a positive integer or zero.');
+    }
+    $this->expireSeconds = $seconds;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpireWarnSeconds(int $seconds): PasswordPolicyInterface {
+    if ($seconds < 0) {
+      throw new PolicyInvalidArgumentException('The expiry warning in seconds must be a positive integer or zero.');
+    }
+    $this->expireWarnSeconds = $seconds;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setExpiryWarningMessage(?string $message): PasswordPolicyInterface {
+    $this->expiryWarningMessage = $message;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setMinimumRequiredConstraints(int $number): PasswordPolicyInterface {
+    if ($number < 0) {
+      throw new PolicyInvalidArgumentException('The minimum requirements cannot be less than zero.');
+    }
+    $this->minimumRequiredConstraints = $number;
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConstraint($constraint): PasswordConstraintInterface {
+    return $this->getConstraints()->get($constraint);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConstraints(): PasswordConstraintPluginCollection {
+    if (!$this->constraintsCollection) {
+      $this->constraintsCollection = new PasswordConstraintPluginCollection($this->getPasswordConstraintPluginManager(), $this->constraints);
+    }
+    return $this->constraintsCollection;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function deletePasswordConstraint(PasswordConstraintInterface $constraint): PasswordPolicyInterface {
+    $this->getConstraints()->removeInstanceId($constraint->getUuid());
+    $this->save();
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addPasswordConstraint(array $configuration): string {
+    $configuration['uuid'] = $this->uuidGenerator()->generate();
+    $this->getConstraints()->addInstanceId($configuration['uuid'], $configuration);
+    return $configuration['uuid'];
+  }
+
+  /**
+   * Returns the password constraint plugin manager.
+   *
+   * @return \Drupal\Component\Plugin\PluginManagerInterface
+   *   The password constraint plugin manager.
+   */
+  protected function getPasswordConstraintPluginManager() {
+    // @FIXME Dependency injection (if possible).
+    return \Drupal::service('plugin.manager.password_enhancements.constraint');
+  }
+
+  /**
+   * Gets the UUID generator.
+   *
+   * @return \Drupal\Component\Uuid\UuidInterface
+   *   The UUID generator service.
+   */
+  protected function uuidGenerator() {
+    // @FIXME Dependency injection (if possible).
```

Comment:
The only possible way to do that is removing query and command methods from a POJO object.

Reply:
Well, as this is more than a POPO, this comment remains there for the time being.